pynamodb/models.pyi

```diff
@@ -52,7 +52,7 @@ class Model(metaclass=MetaModel):
     @overload
     def save(self, condition: Optional[Any] = ..., in_transaction: None = ...) -> Dict[str, Any]: ...
     @overload
-    def save(self, condition: Optional[Any] = ..., in_transaction: TransactWrite = ...) -> None: ...
+    def save(self, in_transaction: TransactWrite, condition: Optional[Any] = ...) -> None: ...
```

Comment:
Uh, the reordering makes it wrong, though.
Maybe
```python
@overload
def save(self, condition: Optional[Any], in_transaction: TransactWrite) -> None: ...
@overload
def save(self, *, condition: Optional[Any] = ..., in_transaction: TransactWrite) -> None: ...
```

Although, maybe we'll just follow the pattern where all those methods are part of the `TransactWrite`?

Reply:
If we're going to follow the same pattern of all the methods being on `TransactWrite`, I would rather us go ahead and make that decision, rather than continue to go back and forth on it. :/

I personally prefer the currently implemented method, as I think its a lot cleaner. As I stated well over a month ago at this point, my team's goal was to have this change merged into pynamodb by the end of this month, and I would greatly prefer to not have to completely rewrite this if there's no real reason to do so.

Reply:
(also, the `*, in_transaction: TransactWrite` suggestion worked, so its passing mypy again)

Reply:
Hallie, it's erroneous with just 2 overloads, though, since `foo.save(condition, my_transaction)` is not unsupported. I intentionally split it into
1. a signature with a non-defaulted `condition`
2. a signature with no `condition` where you must pass `my_transaction` by keyword

(Thankfully we have just 2 arguments here, otherwise it'd quickly get unwieldy.)

Reply:
changed the implementation to be on the `TransactWrite` class